readme updates (#503)
hashicorp/terraform-plugin-framework: README.md

# Terraform Plugin Framework

terraform-plugin-framework is a module for building [Terraform providers](https://www.terraform.io/language/providers). It is built on [terraform-plugin-go](https://github.com/hashicorp/terraform-plugin-go). It aims to provide as much of the power, predictability, and versatility of terraform-plugin-go as it can while abstracting away implementation details and repetitive, verbose tasks.

## Getting Started

> Developers can get started to build the providers using our new [HashiCorp Learn collections](https://developer.hashicorp.com/terraform/tutorials/providers/plugin-framework-create) or upgrade their existing provider using our [migration guide](https://www.terraform.io/plugin/framework/migrating). 

Reviewer: The link for the Learn collection should be https://learn.hashicorp.com/collections/terraform/providers-plugin-framework once it's live 👍 

I also wonder if this might be better suited as a list, similar to https://www.terraform.io/plugin/framework#get-started, which was recommended from the technical writing team. e.g.

```markdown
* Try the [Terraform Plugin Framework collection](https://learn.hashicorp.com/collections/terraform/providers-plugin-framework) on HashiCorp Learn.
* Clone the [terraform-provider-scaffolding-framework](https://github.com/hashicorp/terraform-provider-scaffolding-framework) template repository on GitHub for new providers.
* Follow the [Terraform Plugin Framework migration guide](https://www.terraform.io/plugin/framework/migrating) for converting existing terraform-plugin-sdk providers.
* Read the [Terraform Plugin Framework website](https://www.terraform.io/plugin/framework) for full documentation.
* Browse the [terraform-plugin-framework module](http://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework) on the Go package documentation website.
* Ask questions in the [Terraform Plugin Development section](https://discuss.hashicorp.com/c/terraform-providers/tf-plugin-sdk/43) on HashiCorp Discuss.
```

Learn more about [Terraform Plugin Framework](https://www.terraform.io/plugin/framework).

## Status

terraform-plugin-framework has reached **Public Beta** phase. We are committed to moving forward with the module, but cannot guarantee any of its interfaces will not change as long as it is in version 0. We're waiting for additional feedback, usage, and maturity before we're comfortable committing to APIs with the same years-long support timelines that [terraform-plugin-sdk](https://github.com/hashicorp/terraform-plugin-sdk) brings. We do not expect practitioner experiences to break or change as a result of these changes, only the abstractions surfaced to provider developers.

> terraform-plugin-framework is also not at full feature parity with terraform-plugin-sdk yet. See [Which SDK Should I Use?](https://terraform.io/docs/plugin/which-sdk.html) for more information.

Reviewer: It _is_ at full parity now from a major feature perspective. 😉 

```suggestion
Refer to [Which SDK Should I Use?](https://terraform.io/docs/plugin/which-sdk.html) for more information.
```

> We believe terraform-plugin-framework is a suitable and reliable module to build Terraform providers on, and encourage community members that can afford occasional breaking changes to build with it. terraform-plugin-framework will eventually become GA with a new major version release, at which point its interfaces will be stable, but we need real-world use and feedback before we can be comfortable making those commitments. 

Reviewer: Nit: Prefer spelling out acronyms unless previously defined with parenthesis

```suggestion
We believe terraform-plugin-framework is a suitable and reliable module to build Terraform providers on, and encourage community members that can afford occasional breaking changes to build with it. terraform-plugin-framework will eventually become generally available with a new major version release, at which point its interfaces will be stable, but we need real-world use and feedback before we can be comfortable making those commitments. 
```
